lib/formatters/stringFormatter.js

```diff
@@ -83,9 +84,14 @@ function invalidOptionsFormatter(results) {
  * @return {string}
  */
 function logFrom(fromValue, cwd) {
-	if (fromValue.startsWith('<')) return fromValue;
+	if (fromValue.startsWith('<')) {
+		return underline(fromValue);
+	}
+
+	const filePath = path.relative(cwd, fromValue).split(path.sep).join('/');
+	const fileUrl = `file://${fromValue}`;
 
-	return path.relative(cwd, fromValue).split(path.sep).join('/');
+	return terminalLink(filePath, fileUrl, { fallback: false });
```

Comment:
[note] If not disabling `fallback`, file paths would be output into an unsupported terminal like this:

```console
$ bin/stylelint.js /tmp/*.css --config /tmp/.stylelintrc.yml > out.txt

$ cat out.txt

../../../../../tmp/a.css (file:///tmp/a.css)
 1:2  ✖  Unexpected empty block  block-no-empty

../../../../../tmp/b.css (file:///tmp/b.css)
 1:2  ✖  Unexpected empty block  block-no-empty
```

This may be a breaking change.

Reply:
What do you think if we disable fallback and use https://github.com/jamestalmage/supports-hyperlinks (dependency of terminal-links) to add hyperlinks if they supported?

Reply:
@hudochenkov Thanks for the feedback!
We can use directly `supports-hyperlinks` instead of `terminal-link`!
If doing so, we need to continue to use `ansiEscapes.link()` or write a similar logic by ourselves.

E.g.

```js
function ourTerminalLink(text, url) {
  if (supportsHyperlinks.stdout) {
    return ansiEscapes.link(text, url)
    // or 
    // return ourAnsiEscapeLink(text, url)
  } else {
    return text
  }
}
```

`ansiEscapes.link()` is a tiny utility, so the idea of writing similar logic sounds good, too. (we don't need to depend on old versions of the packages)

- https://github.com/sindresorhus/terminal-link/blob/a925ecd9f2e6975505d801b5e352a05a9ff5921a/index.js#L14
- https://github.com/sindresorhus/ansi-escapes/blob/2b3b59c56ff77a2afdee946bff96f1779d10d775/index.js#L92-L107

Or, it seems good to use the `hyperlinker` package instead of `ansiEscapes.link()`.

https://github.com/jamestalmage/hyperlinker

What do you think?

Reply:
> ansiEscapes.link() is a tiny utility, so the idea of writing similar logic sounds good, too. (we don't need to depend on old versions of the packages)

This sounds good to me.